modules/programs/alacritty/default.nix

```diff
@@ -0,0 +1,234 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  defaults = import ./default-config.nix;
+  cfg = config.programs.alacritty;
+
+  mapAttrNamesRecursive' = f: set:
+    let
+      recurse = set:
+        let
+          g =
+            name: value:
+            if isAttrs value
+              then nameValuePair (f name) (recurse value)
+              else nameValuePair (f name) value;
+        in mapAttrs' g set;
+    in recurse set;
+
+  toSnakeCase = replaceChars upperChars (map (c: "_${c}") lowerChars);
+
+  animationList = [
+    "Ease"
+    "EaseOut"
+    "EaseOutSine"
+    "EaseOutQuad"
+    "EaseOutCubic"
+    "EaseOutQuart"
+    "EaseOutQuint"
+    "EaseOutExpo"
+    "EaseOutCirc"
+    "Linear"
+  ];
+  actionList = [
+    "Paste"
+    "PasteSelection"
+    "Copy"
+    "IncreaseFontSize"
+    "DecreaseFontSize"
+    "ResetFontSize"
+    "ScrollPageUp"
+    "ScrollPageDown"
+    "ScrollToTop"
+    "ScrollToBottom"
+    "ClearHistory"
+    "Hide"
+    "Quit"
+    "ClearLogNotice"
+  ];
+  modList = [
+    "Command"
+    "Control"
+    "Shift"
+    "Alt"
+  ];
+  modeList = [
+    "~AppCursor"
+    "AppCursor"
+    "~AppKeypad"
+    "AppKeypad"
+  ];
+in {
+  options.programs.alacritty = with types; {
+    enable = mkEnableOption "Alacritty terminal";
+    window = {
+      dimensions = {
+        columns = mkOption {
+          type = int;
+          default = 80;
+          description = "Number of columns in the window";
+        };
+        lines = mkOption {
+          type = int;
+          default = 24;
+          description = "Number of lines in the window";
+        };
+      };
+      padding = {
+        x = mkOption {
+          type = int;
+          default = 2;
+          description = "Horizontal padding around the window, in pixels";
+        };
+        y = mkOption {
+          type = int;
+          default = 2;
+          description = "Vertical padding around the window, in pixels";
+        };
+      };
+      decorations = mkOption {
+        description = ''
+          Window decorations
```

Comment:
Recall that descriptions are handled in docbook format. So this would show up as

![2019-01-13-013142_711x182_scrot](https://user-images.githubusercontent.com/798147/51079971-08058200-16d3-11e9-949d-866e66eb6e27.png)

in the documentation. I think a description list would be the best choice:

```xml
Window decorations
<variablelist>
  <varlistentry>
    <term><literal>none</literal></term>
    <listitem>
      <para>Neither borders nor title bar.</para>
    </listitem>
  </varlistentry>
  <varlistentry>
    <term><literal>full</literal></term>
    <listitem>
      <para>Borders and title bar.</para>
    </listitem>
  </varlistentry>
</variablelist>
```

and this shows up as

![2019-01-13-012944_341x295_scrot](https://user-images.githubusercontent.com/798147/51079970-08058200-16d3-11e9-858d-7f89266e1584.png)